packages/react-instantsearch-hooks/src/hooks/useInstantSearch.ts

```diff
@@ -0,0 +1,56 @@
+import { useCallback } from 'react';
+
+import { useInstantSearchContext } from '../lib/useInstantSearchContext';
+import { useSearchResults } from '../lib/useSearchResults';
+import { useSearchState } from '../lib/useSearchState';
+
+import type { SearchResults } from 'algoliasearch-helper';
+import type {
+  IndexUiState,
+  InstantSearch,
+  Middleware,
+  ScopedResult,
+  UiState,
+} from 'instantsearch.js';
+
+type InstantSearchApi = {
+  scopedResults: ScopedResult[];
+  results: SearchResults<any>;
+  uiState: UiState;
+  setUiState: InstantSearch['setUiState'];
+  indexUiState: IndexUiState;
+  setIndexUiState: (indexUiState: IndexUiState) => void;
+  use: (...middlewares: Middleware[]) => void;
+  refresh: InstantSearch['refresh'];
+  search: InstantSearch;
```

Comment:
We don't want to provide access to `search` in this API until it's proven to be useful, which it shouldn't.

Reply:
I feel like there's other keys that don't need to be "exposed as much", but can still be useful, like `search.renderState` for accessing refinement functions without mounting that widget again. If we feel very strongly it can be removed again, but I saw the use in it